### PR TITLE
simulators/common: remove local provider

### PR DESCRIPTION
**Pull request by renaynay**
This PR removes the option to add a local provider as this functionality is largely unused and adds complexity to adding new test suites / simulations.

Now, it is possible to get the hive host simply by calling 

```go
host, err := hive.New()
```

### CODE REVIEW

File: simulators/common/providers/hive/hive.go [hunk 1]
```diff
@@ -33,9 +33,11 @@ type host struct {
...
 var once sync.Once
 
-// Support this provider type to register it
-func Support() {
-	common.RegisterProvider("hive", GetInstance)
+// New takes the value of the environment variable "HIVE_SIMULATOR"
+// and returns a new common.TestSuiteHost
+func New() (common.TestSuiteHost, error) {
+	config := []byte(fmt.Sprintf(`{"hostURI":%q}`, os.Getenv("HIVE_SIMULATOR")))
+	return GetInstance(config)
```
Comment (fjl) [inline simulators/common/providers/hive/hive.go:40]: Please add a check which panics if the HIVE_SIMULATOR environment variable is not set.
You can also simplify this by just constructing the HostConfiguration and host objects directly.

**Review by fjl — CHANGES_REQUESTED** [review]

File: simulators/common/providers/hive/hive.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"github.com/ethereum/hive/simulators/common"
```
Comment (holiman) [inline simulators/common/providers/hive/hive.go:7]: goimports

File: simulators/ethereum/graphql/graphql.go [hunk 3]
```diff
@@ -3,8 +3,8 @@ package main
 import (
 	"bytes"
 	"encoding/json"
-	"flag"
 	"fmt"
+	"github.com/ethereum/hive/simulators/common/providers/hive"
```
Comment (holiman) [inline simulators/ethereum/graphql/graphql.go:7]: goimports :)

File: simulators/ethereum/sync/sync_test.go [hunk 4]
```diff
@@ -2,8 +2,8 @@ package main
 
 import (
 	"context"
-	"flag"
 	"fmt"
+	"github.com/ethereum/hive/simulators/common/providers/hive"
```
Comment (holiman) [inline simulators/ethereum/sync/sync_test.go:6]: ...

**Review by holiman — APPROVED** [review]
`+31, -722` -- gotta love it!

**Pull request merged**